test/e2e/trafficroute/universal_multizone/traffic_route.go

```diff
@@ -0,0 +1,295 @@
+				WithServiceVersion("v1"),
+				WithTransparentProxy(true),
+			)).
+			Install(IngressUniversal(defaultMesh, ingressToken)).
+			Setup(remote_2)
+		Expect(err).ToNot(HaveOccurred())
+		err = remote_2.VerifyKuma()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	E2EAfterEach(func() {
+		// remove all TrafficRoutes
+		items, err := global.GetKumactlOptions().KumactlList("traffic-routes", "default")
+		Expect(err).ToNot(HaveOccurred())
+		for _, item := range items {
+			if item == "route-all-default" {
+				continue
+			}
+			err := global.GetKumactlOptions().KumactlDelete("traffic-route", item, "default")
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	E2EAfterSuite(func() {
+		Expect(remote_1.DeleteKuma(optsRemote1...)).To(Succeed())
+		Expect(remote_1.DismissCluster()).To(Succeed())
+
+		Expect(remote_2.DeleteKuma(optsRemote2...)).To(Succeed())
+		Expect(remote_2.DismissCluster()).To(Succeed())
+
+		Expect(global.DeleteKuma(optsGlobal...)).To(Succeed())
+		Expect(global.DismissCluster()).To(Succeed())
+	})
+
+	collectResponses := func(url string, expected, notExpected []string) {
+		instances := map[string]bool{}
+		Eventually(func() bool {
+			stdout, _, err := remote_1.ExecWithRetries("", "", "demo-client",
+				"curl", "-v", "-m", "3", "--fail", url)
+			if err != nil {
+				return false
+			}
+
+			for _, ne := range notExpected {
+				Expect(stdout).ToNot(ContainSubstring(ne))
+			}
+
+			for _, e := range expected {
+				if strings.Contains(stdout, e) {
+					instances[e] = true
+				}
+			}
+
+			return len(instances) == len(expected)
+		}, "30s", "500ms").Should(BeTrue())
+	}
+
+	within := func(n float64, l, r int) bool {
+		return float64(l) <= n && n <= float64(r)
+	}
+
+	waitRatio := func(url string, v1Weight, v2Weight int, echo1, echo2 string) error {
+		minAttempts := 100
+		maxAttempts := 1000
+		errorPercent := 3
+
+		Eventually(func() bool {
+			_, _, err := remote_1.ExecWithRetries("", "", "demo-client",
+				"curl", "-v", "-m", "3", "--fail", url)
+			return err == nil
+		}, "30s", "500ms").Should(BeTrue())
+
+		echo1Resp := float64(0)
+		echo2Resp := float64(0)
+		for i := 0; i < maxAttempts; i++ {
+			stdout, _, err := remote_1.ExecWithRetries("", "", "demo-client",
+				"curl", "-m", "3", "--fail", url)
+			Expect(err).ToNot(HaveOccurred())
+
+			if strings.Contains(stdout, echo1) {
+				echo1Resp++
+			}
+			if strings.Contains(stdout, echo2) {
+				echo2Resp++
+			}
+
+			if i < minAttempts {
+				continue
+			}
+
+			currentPercent1 := echo1Resp / float64(i+1) * 100
+			currentPercent2 := echo2Resp / float64(i+1) * 100
+
+			if within(currentPercent1, v1Weight-errorPercent, v1Weight+errorPercent) && within(currentPercent2, v2Weight-errorPercent, v2Weight+errorPercent) {
+				return nil
+			}
+		}
+		return errors.Errorf("resulted split values %v and %v aren't within original values %v and %v",
+			echo1Resp/float64(maxAttempts)*100, echo2Resp/float64(maxAttempts)*100, v1Weight, v2Weight)
+	}
```

Comment:
What is the advantage of this algo vs simple send 100 requests, see how they are distributed?